awssqs/pkg/apis/sources/v1alpha1/aws_sqs_types.go

```diff
@@ -119,12 +116,19 @@ func (s *AwsSqsSourceStatus) InitializeConditions() {
 
 // MarkSink sets the condition that the source has a sink configured.
 func (s *AwsSqsSourceStatus) MarkSink(uri string) {
-	s.SinkURI = uri
 	if len(uri) > 0 {
-		condSet.Manage(s).MarkTrue(AwsSqsSourceConditionSinkProvided)
+		if u, err := apis.ParseURL(uri); err != nil {
+			s.SinkURI = nil
+			condSet.Manage(s).MarkUnknown(AwsSqsSourceConditionSinkProvided,
+				"SinkEmpty", "Sink resolving resulted in an error. %s", err.Error())
```

Comment:
This is a bug fix right? Does it change the behavior when a referenced sink has an invalid addressable uri? If so add it to release notes.

Reply:
it should be exactly the same.

Reply:
and if it was important there would be a test. Tests passed.

Reply:
Users don't care about tests or not, they care about what changed. We have lots of untested behavior. But if you're sure it's the same, no need for a release note.

Reply:
I am going to delete the sdk. NOW.